src/utils/buildMetadata.ts

```diff
@@ -166,13 +153,198 @@ export function buildMetadata(sheets: Sheet[], defaultCC: string) {
         optionSets,
         trackedEntityAttributes,
         trackedEntityTypes,
-        //programs,
+        programSections: buildprogramSections(sheets),
+        programs: buildPrograms(sheets),
+        programStages: buildProgramStages(sheets),
+        programStageSections: buildProgramStageSections(sheets),
         programRules: buildProgramRules(sheets),
         programRuleActions: buildProgramRuleActions(sheets),
         programRuleVariables: buildProgramRuleVariables(sheets),
     };
 }
 
+function buildPrograms(sheets: Sheet[]) {
+    const get = (name: string) => getItems(sheets, name);
+
+    const programs = get("programs");
+    const pSections = get("programSections");
+    const pStages = get("programStages");
+    const trackedEntityTypes = get("trackedEntityTypes");
+    const categoryCombos = get("categoryCombos");
+
+    return programs.map(program => {
+        const trackedEntityType = {
+            id: getByName(trackedEntityTypes, program.trackedEntityType)?.id
+        };
+
+        const programStages = pStages.filter((programStages) => {
+            return programStages?.program === program.name;
```

Comment:
(This happens in similar functions below, by the way, but I imagine you know that!)

Reply:
Its named programStages because that's the name that is used in the program JSON schema.
Also, that variable stores the programStages associated to the program, so it can store one or more ids (tracker programs can have multiple stages).

You right about the fallback undefined for programStage.

About the functions, that may be me being a noob but i prefer the more explicit version with {} and return, i find it more readable.

Reply:
About this:
```
makeSortOrder(programStageSections.filter((programStageSections) => {
    return programStageSections.program === programStageSection.program &&
        programStageSections.programStage === programStageSection.programStage
```
Its filtering names here, so it checks that both program and programStage its the same.
A different program can have a programStage with a repeated name.